### PR TITLE
WIP Clarify limitations of Azure File storage

### PR DESCRIPTION
**Pull request by jboxman**
- https://bugzilla.redhat.com/show_bug.cgi?id=1601757

### CODE REVIEW

**Comment by jboxman** [comment]
@openshift/team-documentation, I'm interested in any suggestions on how better to structure this content. The previous update was triggered by this[0] bug. Thanks.

[0] https://bugzilla.redhat.com/show_bug.cgi?id=1601757

File: install_config/persistent_storage/persistent_storage_azure_file.adoc [hunk 1]
```diff
@@ -54,10 +54,29 @@ the default `dir_mode` and `file_mode` permissions changed from `0777` to `0755`
 in {product-title} 3.9. The following examples show configuration files with the
 changed values.
 
-.Considerations when using MySQL and PostgreSQL with Azure file
-* The owner UID of the Azure File mounted directory is different from the UID of a container.
-* MySQL containers change the file owner permission in the mounted directory. Because of the mismatch between the owner UID and container process UID, this operation fails. Therefore to run MySQL with Azure File:
-** Specify the Azure File mounted directory UID in the `runAsUser` variable in the PV configuration file.
+.Considerations when using Azure File
```
Comment (kalexand-rh) [inline install_config/persistent_storage/persistent_storage_azure_file.adoc]: Lines 57 to 68 sound move like general restrictions than prereqs. I might move them to line 27.

File: install_config/persistent_storage/persistent_storage_azure_file.adoc [hunk 2]
```diff
@@ -54,10 +54,29 @@ the default `dir_mode` and `file_mode` permissions changed from `0777` to `0755`
...
+
+The following file system features are not supported by Azure File:
+
+* Symlinks
+* Hard links
+* Extended attributes
+* Sparse files
+* Named pipes
+
+Additionally, the owner UID of the Azure File mounted directory is different
```
Comment (kalexand-rh) [inline install_config/persistent_storage/persistent_storage_azure_file.adoc]: Please expand UID.

File: install_config/persistent_storage/persistent_storage_azure_file.adoc [hunk 3]
```diff
@@ -54,10 +54,29 @@ the default `dir_mode` and `file_mode` permissions changed from `0777` to `0755`
...
+from the process UID of the container.
+
+You may experience instability in your environment if you use any container
```
Comment (kalexand-rh) [inline install_config/persistent_storage/persistent_storage_azure_file.adoc]: s/may/might

File: install_config/persistent_storage/persistent_storage_azure_file.adoc [hunk 4]
```diff
@@ -54,10 +54,29 @@ the default `dir_mode` and `file_mode` permissions changed from `0777` to `0755`
...
+images that utilize unsupported file system features. Containers for PostgreSQL
```
Comment (kalexand-rh) [inline install_config/persistent_storage/persistent_storage_azure_file.adoc]: s/utilize/use

File: install_config/persistent_storage/persistent_storage_azure_file.adoc [hunk 5]
```diff
@@ -54,10 +54,29 @@ the default `dir_mode` and `file_mode` permissions changed from `0777` to `0755`
...
+and MySQL are known to have issues when used with Azure File.
+
+For MySQL containers, the container changes the file owner permission in the
+mounted directory. Because of the mismatch between the owner UID and container
+process UID, this operation fails. The following workaround is provided:
```
Comment (kalexand-rh) [inline install_config/persistent_storage/persistent_storage_azure_file.adoc]: What operation?

File: install_config/persistent_storage/persistent_storage_azure_file.adoc [hunk 6]
```diff
@@ -54,10 +54,29 @@ the default `dir_mode` and `file_mode` permissions changed from `0777` to `0755`
...
+* Named pipes
+
+Additionally, the owner UID of the Azure File mounted directory is different
+from the process UID of the container.
+
+You may experience instability in your environment if you use any container
+images that utilize unsupported file system features. Containers for PostgreSQL
+and MySQL are known to have issues when used with Azure File.
+
+For MySQL containers, the container changes the file owner permission in the
```
Comment (kalexand-rh) [inline install_config/persistent_storage/persistent_storage_azure_file.adoc]: I'd rewrite this section to start with ". If you use MySQL containers, you must modify the PV configuration file..."
I might use two substeps for the different modifications. I'd also clarify what the two sample files (.Storage class configuration file example and .PV configuration file example) are for.

File: install_config/persistent_storage/persistent_storage_azure_file.adoc [hunk 7]
```diff
@@ -29,7 +29,7 @@ storage provider.
 == Before you begin
 . Install `samba-client`, `samba-common`, and `cifs-utils` on all nodes:
 +
-[source, bash]
+[source,bash]
```
Comment (kalexand-rh) [inline install_config/persistent_storage/persistent_storage_azure_file.adoc]: We decided to drop [source,bash] in the 4.0 stream, so consider removing it from this file.

**Review by kalexand-rh — COMMENTED** [review]
A few picks and suggestions. I'd make sure that the prereqs are all really prereqs and not conceptual material.

**Comment by jboxman** [comment]
Re-opened as https://github.com/openshift/openshift-docs/pull/15261.

**Pull request closed without merging**